src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -760,22 +761,28 @@ private UsingDirectiveSyntax ParseUsingDirective()
                 //NB: there's no way this could be true for a set of tokens that form a valid 
                 //using directive, so there's no danger in checking the error case first.
 
-                name = WithAdditionalDiagnostics(CreateMissingIdentifierName(), GetExpectedTokenError(SyntaxKind.IdentifierToken, this.CurrentToken.Kind));
+                type = WithAdditionalDiagnostics(CreateMissingIdentifierName(), GetExpectedTokenError(SyntaxKind.IdentifierToken, this.CurrentToken.Kind));
                 semicolon = SyntaxFactory.MissingToken(SyntaxKind.SemicolonToken);
             }
             else
             {
-                name = this.ParseQualifiedName();
-                if (name.IsMissing && this.PeekToken(1).Kind == SyntaxKind.SemicolonToken)
+                type = alias == null ? this.ParseQualifiedName() : this.ParseType();
+                if (type is not NameSyntax)
+                    type = CheckFeatureAvailability(type, MessageID.IDS_FeatureUsingTypeAlias);
+
+                if (type is RefTypeSyntax)
+                    type = AddError(type, ErrorCode.ERR_BadRefInUsingAlias);
```

Comment:
fairly certain this is appropriate.  there are no real 'ref types' (even though we designed the parser to allow that).  Instead, there are contexts where 'ref' is allowed (like in a local declaration statement).